packages/compiler/src/compiler/base/nodes/each.ts

```diff
@@ -54,14 +73,22 @@ export async function compile({
       }
       localScope.set(indexVarName, indexValue)
     }
+
     for await (const childNode of node.children ?? []) {
       await resolveBaseNode({
         node: childNode,
         scope: localScope,
         isInsideMessageTag,
         isInsideContentTag,
+        completedValue: `step_${i}`,
       })
     }
+
     i++
   }
+
+  nodeWithStatus.status = {
+    ...nodeWithStatus.status,
+    loopIterationIndex: 0,
+  }
```

Comment:
The `{{#each …}}` tag stores additional information in its status to know which iterations to skip when performing a new step.